jenkins/bootstrap.py

```diff
@@ -330,10 +330,23 @@ def cat(self, path, generation):
         return self.call(cmd, output=True)
 
 
-    def upload_artifacts(self, path, artifacts):
+    def upload_artifacts(self, gsutil, path, artifacts):
         """Upload artifacts to the specified path."""
         # Upload artifacts
         if os.path.isdir(artifacts):
+            try:
+                # If remote path exists, it will create .../_artifacts subdir instead
+                gsutil.stat(path)
```

Comment:
I guess so. It runs as part of log-dump.sh, so it either runs inside k8s e2e-framework as afterSuite, or inside kubetest DumpClusterLogs() function.